xalia/Win32/AccessibleProvider.cs

```diff
@@ -917,6 +962,27 @@ public static ElementIdentifier ElementIdFromVariantBackground(object variant, I
                 return result;
             }
 
+            if (!(uiaprov is null)) {
+                var fragment = uiaprov as IRawElementProviderFragment;
+                if (!(fragment is null))
+                {
+                    var runtime_id = fragment.GetRuntimeId();
+
+                    if (!(runtime_id is null))
+                    {
+                        if (runtime_id[0] == UiaAppendRuntimeId)
+                        {
+                            result.runtime_id = new int[runtime_id.Length + 2];
+                            result.runtime_id[0] = 42;
+                            result.runtime_id[1] = (int)root_hwnd;
+                            Array.Copy(runtime_id, 0, result.runtime_id, 2, runtime_id.Length);

```

Comment:
It probably isn't a big deal, but it looks to me like you're copying the `UiaAppendRuntimeId` value here into your `result.runtime_id` array. It won't hurt anything, but it's just adding an extra number that doesn't differentiate things anymore than the starting `42` value does. Native usually puts a value in that place that indicates the type of the fragment root for the provider, e.g if the fragment root has `ProviderOptions_NonClientAreaProvider`, the value is 3, if it's an override provider it's 5, and it defaults to 4.

Reply:
This is duplicated logic, not new logic, but it actually does matter. We need this to match what we calculate from the runtime id we get in OnUiaEvent.

Reply:
So probably Xalia would only ever get 4 for this, right? Since it goes out of its way to only use server-side providers.

Reply:
That logic also should probably be moved into GetElementName so we don't have to duplicate it.

Reply:
Yeah, it should only ever get 4. In theory someone could have their remote provider marked as nonclient, but I don't think I've ever seen that and I don't expect it to happen.

Reply:
Added a commit to fix this, seem good to you?

Reply:
Yes, looks good to me. :)